### ISSUE TITLE
Prevent escaping of strings on variable substitution

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1126]
In GitLab by @abaer-kit on Sep 1, 2023, 17:53

Hello @dbeltrankyl and @kinow,

### Summary of the feature

If I understand the behavior correctly, changing from AS v4.0.84 to v4.0.86, variable substitution seems to have changed in the sense that all string variables are now replaced with special characters escaped, i.e. `FILE: "${MYPATH}/some-file.dat"` will result in `%FILE%` to be replaced with `\$\{MYPATH\}/some\-file\.dat`. I see that this is reasonable for some cases (substitutions in a bash script do not have to be quoted), but it prevents the use of bash environment variables without the `eval` construct.

I would appreciate the option to turn this off, ideally on a single field basis.
Furthermore some documentation, e.g. which characters are escaped would be great.

### Describe your use case

I have two use cases:
1. Construction of input/output paths. I am loading several different files with configuration snippets to construct the full input and output paths, e.g.
Platform-specific file:
```yaml
DIRECTORIES:
  PREFIX_OUTDIR: '$SCRATCH'
```
Experiment-specific file:
```yaml
EXPNAME: myexp
DIRECTORIES:
  INDIR:  "%DIRECTORIES.PREFIX_INDIR%/input/%EXPNAME%"
```
The platform-specific prefix can on some systems easily be constructed with a bash environment variable (that might be pre-defined on the system).

2. In the other case, the time step is calculated, depending on the resolution of an LES simulation and then inserted into the template (which itself modifies the experiment namelist):
```yaml
TIMESTEP: '$(echo "scale=3; (%CUSTOM.RESOLUTION% * 0.0072) / 1" | bc)'
```

### Describe the problem

With all special characters escaped, bash treats these as strings and does not evaluate the content. As this is fine and for sure useful for many cases, it would be great to have the option to control it through the configuration, i.e. allow the user to insert something "to be evaluated" or something "constant".

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 4, 2023, 08:00

Hello @abaer-kit ,

Sorry, this was fixed in 4.0.87 but by mistake it was not uploaded into pip, please use that version it should now be available

**Comment by kinow** [comment]
In GitLab by @abaer-kit on Sep 4, 2023, 15:45

Yes, v4.0.87 fixes it. Thank you!
